Adjust ThreadListStackTracesTest for checkReentrantLock

singh264 opened the pull request:
Adjust ThreadListStackTracesTest for checkReentrantLock
to be ready when the virtual thread state is WAITING after
the reentrant lock parks the virtual thread. When the
virtual thread is mounted, the virtual thread state, which
is obtained from the carrier thread, can transition
between WAITING and RUNNABLE before the virtual thread is
parked on the lock.

Issue: https://github.com/eclipse-openj9/openj9/issues/18089
Signed-off-by: Amarpreet Singh <Amarpreet.A.Singh@ibm.com>

singh264 commented:
@babsingh requesting review

babsingh commented:
Feedback for https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/63 also applies to this PR.

singh264 commented:
> Feedback for https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/63 also applies to this PR.

@babsingh I have addressed your feedback, re-requesting your review

singh264 commented:
@babsingh I have squashed the commits, updated the commit message, and addressed the comments at https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/63 in this PR.

Re-requesting your review.

babsingh reviewed (approved):
300x grinder passed: https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/36164.
